pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1347,6 +1350,23 @@ func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string)
 						log.Errorf("failed to create wcpCapabilityApi client. Err: %+v", err)
 						return false
 					}
+					// Check if CRD for capabilities exists
+					// If CRD does not exist on supervisor then skip futher capability check
+					// this is case when tkr is newer and supervisor is older where capabilities CRD does not exists
+					apiextensionsClientSet, err := apiextensionsclientset.NewForConfig(restClientConfig)
+					if err != nil {
+						log.Errorf("failed to create apiextension clientset using config. Err: %+v", err)
+						return false
+					}
+					_, err = apiextensionsClientSet.ApiextensionsV1().CustomResourceDefinitions().Get(ctx,
+						"capabilities.iaas.vmware.com", metav1.GetOptions{})
+					if err != nil {
```

Comment:
For workload-isolation-domain feature, we can use `return c.IsCNSCSIFSSEnabled(ctx, featureName)`, as capability CRD is not registered on 8.x supervisor.